source/compiler-core/slang-artifact-handler-impl.cpp

```diff
@@ -29,8 +29,9 @@ namespace Slang {
 
 SlangResult DefaultArtifactHandler::queryInterface(SlangUUID const& uuid, void** outObject)
 {
-	if (auto ptr = getInterface(uuid))
+	if ([[maybe_unused]] auto ptr = getInterface(uuid))
```

Comment:
This looks like a bug in the original code. *outObject should just return ptr. 
ptr doesn't have to be equal to this, so that should be removed.

Reply:
Yeah, overall this bit was a little suspicious. The reason I did it this way was that getInterface doesn't increment the reference count (as it should according to the COM documentation. So the only correct and behavior preserving fix was to make sure that this was the same pointer.